index.html

```diff
@@ -3,47 +3,256 @@
+	{"state":"Wisconsin","employment":2936.2,"region":"Midwest"},
+	{"state":"Wyoming","employment":284.4,"region":"West"}
+  ];
+
+  var MAX_EMPLOYMENT = 16322;
+
+  var options = {
+    sorted: false,
+    filtered: false,
+    page: 0
+  };
+    
+    
+    var tbody = d3.select("tbody");
+
+  function updateTable(){
+    var rows = "";
+    tbody.html("");
+    var data = states.slice();
+      
+    if (options.sorted) {
+      if (options.sorted === "asc") {
+          data.sort(function (a, b) { return a.employment - b.employment; });
+      } else if (options.sorted === "desc") {
+          data.sort(function (a, b) { return b.employment - a.employment; });
+      } 
+    }        
+    
+
+    if (options.filtered) {
+      data = data.filter(function (d) { 
+          return d.region === options.filtered; 
+          });
+    }      
+            
+    rows = tbody.selectAll("tr")
+        .data(data)
+        .enter()
+        .append("tr");
+
+    rows.append("td")
+      .html(function(d) { return d.state });
+      
+    rows.append("td")
+        .html(function(d) { return d.employment });
+
+    rows.append("td")
+      .html(function(d) { return d.region });
+
+    rows.append("td")
+      .append("div")
+      .classed("bar", true)
+      .style("width", function(d) { return ((d.employment / MAX_EMPLOYMENT) * 100) + "%" })
+
+  };   
+    
+  updateTable();
+
+
+// sort ascending 
+    d3.select("#sort-asc").on("click", function() {
+        if (options.sorted === "asc") {
+        options.sorted = false;
+        d3.select("#sort-asc").classed("active", false);    
+        } else {
+            options.sorted = "asc";
+            d3.select("#sort-asc").classed("active", true);
+            d3.select("#sort-desc").classed("active", false);
+        }
+        updateTable();
+    });
+
+// sort descending    
+    d3.select("#sort-desc").on("click", function() {
+        if (options.sorted === "desc") {
+        options.sorted = false;
+        d3.select("#sort-desc").classed("active", false);    
+        } else {
+            options.sorted = "desc";
+            d3.select("#sort-desc").classed("active", true);
+            d3.select("#sort-asc").classed("active", false);
+        }
+        updateTable();
+    });    
+    
+// filter for the 4 regions
+    d3.select("#filter-ne").on("click", function() {
+        if (options.filtered === "Northeast") {
+            options.filtered = false;
+            d3.select("#filter-ne").classed("active", false);                
+        } else {
+            options.filtered = "Northeast";
+            d3.select("#filter-so").classed("active", false);                            
+            d3.select("#filter-mw").classed("active", false);                                        
+            d3.select("#filter-we").classed("active", false);                                        
+            d3.select("#filter-ne").classed("active", true);                            
```

Comment:
Yeah, I had tried that but couldn't get it to work. I added an id="filters"
here:

      <div class="btn-group" id="filters">
        <button type="button" class="btn btn-default"
id="filter-ne">Northeast</button>
        <button type="button" class="btn btn-default"
id="filter-so">South</button>
        <button type="button" class="btn btn-default"
id="filter-mw">Midwest</button>
        <button type="button" class="btn btn-default"
id="filter-we">West</button>

And then did like this. Is that the idea? It didn't work for me.

            d3.selectAll("#filters").classed("active", false);

            d3.select("#filter-mw").classed("active", true);

On Sun, Jun 26, 2016 at 1:02 PM, Chris Given <notifications@github.com>
wrote:

> In index.html
> <https://github.com/JHU-DataViz-Summer16/week-1-html-table/pull/4#discussion_r68505925>
> :
>
> > +            d3.select("#sort-asc").classed("active", false);
> > +        }
> > +        updateTable();
> > +    });
> > +
> > +// filter for the 4 regions
> > +    d3.select("#filter-ne").on("click", function() {
> > +        if (options.filtered === "Northeast") {
> > +            options.filtered = false;
> > +            d3.select("#filter-ne").classed("active", false);
> > +        } else {
> > +            options.filtered = "Northeast";
> > +            d3.select("#filter-so").classed("active", false);
> > +            d3.select("#filter-mw").classed("active", false);
> > +            d3.select("#filter-we").classed("active", false);
> > +            d3.select("#filter-ne").classed("active", true);
>
> Great, this works fine. But a lot of duplicative code, which is a good
> indication that this is an opportunity for refactoring. Here's how I'd do
> that: we know that we almost always want all of the buttons to not have the
> active class. So as the first thing in this function, we could use
> d3.selectAll() to select all of the filter buttons and set the active
> class to false. Now, we only need line 207, setting the active class on the
> clicked button to true if it wasn't already selected.
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/JHU-DataViz-Summer16/week-1-html-table/pull/4/files/8cd90789153508c1824ffc63cc6f6f9e44cf4134#r68505925>,
> or mute the thread
> <https://github.com/notifications/unsubscribe/ARl-CDfBvou7mukJ_bD_KBi0wgIklEAOks5qPrCygaJpZM4Iv8j1>
> .
>



-- 
"Narcissism is the fast food of the soul."
     -- Adbusters magazine